crates/nu-engine/src/documentation.rs

```diff
@@ -27,70 +32,216 @@ pub fn get_full_help(
+    stack: &mut Stack,
+) -> String {
+    let Some(content) = captures.get(1) else {
+        // this shouldn't happen
+        return String::new();
+    };
 
-        if let Ok(output) = decl.run(
-            engine_state,
-            stack,
-            &(&call).into(),
-            Value::string(code_string, Span::unknown()).into_pipeline_data(),
-        ) {
-            let result = output.into_value(Span::unknown());
-            if let Ok(s) = result.and_then(Value::coerce_into_string) {
-                return s; // successfully highlighted string
-            }
-        }
+    // Save current color config
+    let config_old = stack.get_config(engine_state);
+    let mut config = (*config_old).clone();
+
+    // Style externals and external arguments with fallback style,
+    // so nu-highlight styles code which is technically valid syntax,
+    // but not an internal command is highlighted with the fallback style
+    let code_style = Value::record(
+        record! {
+            "attr" => Value::string("di", Span::unknown()),
+        },
+        Span::unknown(),
+    );
+    let color_config = &mut config.color_config;
+    color_config.insert("shape_external".into(), code_style.clone());
+    color_config.insert("shape_external_resolved".into(), code_style.clone());
+    color_config.insert("shape_externalarg".into(), code_style);
+
+    // Apply config with external argument style
+    stack.config = Some(Arc::new(config));
+
+    // Highlight and reject invalid syntax
+    let highlighted = try_nu_highlight(content.into(), true, engine_state, stack);
+
+    // Restore original config
+    stack.config = Some(config_old);
+
+    // Use fallback style if highlight failed/syntax was invalid
+    highlighted.unwrap_or_else(|| highlight_fallback(content.into()))
+}
+
+/// Apply fallback code style
+fn highlight_fallback(text: &str) -> String {
+    format!("{DEFAULT_DIMMED}{DEFAULT_ITALIC}{text}{RESET}")
+}
+
+/// Highlight code within backticks
+///
+/// Will attempt to use nu-highlight, falling back to dimmed and italic on invalid syntax
+fn highlight_code<'a>(
+    text: &'a str,
+    engine_state: &EngineState,
+    stack: &mut Stack,
+) -> Cow<'a, str> {
+    let config = stack.get_config(engine_state);
+    if !config.use_ansi_coloring.get(engine_state) {
+        return Cow::Borrowed(text);
     }
-    code_string.to_string()
+
+    // See [`tests::test_code_formatting`] for examples
+    let pattern = r"(?x)     # verbose mode
+        (?<![\p{Letter}\d])    # negative look-behind for alphanumeric: ensure backticks are not directly preceded by letter/number.
+        `
+        ([^`\n]+?)           # capture characters inside backticks, excluding backticks and newlines. ungreedy.
+        `
+        (?![\p{Letter}\d])     # negative look-ahead for alphanumeric: ensure backticks are not directly followed by letter/number.
+    ";
+
+    let re = Regex::new(pattern).expect("regex failed to compile");
```

Comment:
Constructing regexes is considered pricy according to their docs, so I would inline something like that here:
```suggestion
    static RE: LazyLock<Regex> = LazyLock::new(|| Regex::new(pattern).expect("valid regex"));
```

Reply:
oh great idea

Reply:
added in bafd906ead358d6230c2a4358e67e9738a636789